oval-schemas/oval-definitions-schema.xsd

```diff
@@ -239,6 +240,73 @@
+            </xsd:documentation>
+            <xsd:documentation>The required id attribute is the OVAL-ID of the EncapuslatedDefinition. The form of an OVAL-ID must follow the specific format described by the oval:DefinitionIDPattern. The required version attribute holds the current version of the encapsulated definition. Versions are integers, starting at 1 and incrementing every time a definition is modified. The required class attribute indicates the specific class to which the definition belongs. The class gives a hint to a user so they can know what the encapsulated definition writer is trying to say. See the definition of oval-def:ClassEnumeration for more information about the different valid classes. The optional deprecated attribute signifies that an id is no longer to be used or referenced but the information has been kept around for historic purposes.
+            </xsd:documentation>
+            <xsd:documentation>When the deprecated attribute is set to true, the encapsulated definition is considered to be deprecated. The criteria child element of a deprecated encapsulated definition is optional. If a deprecated encapsulated definition does not contain a criteria child element, the encapsulated definition must evaluate to "not evaluated". If a deprecated encapsulated definition contains a criteria child element, an interpreter should evaluate the encapsulated definition as if it were not deprecated, but an interpreter may evaluate the encapsulated definition to "not evaluated".
+            </xsd:documentation>
+            <xsd:appinfo>
+                <sch:pattern id="oval-def_required_criteria">
+                    <sch:rule context="oval-def:oval_definitions/oval-def:definitions/oval-def:definition[(@deprecated='false' or @deprecated='0') or not(@deprecated)]">
+                        <sch:assert test="oval-def:criteria">A valid OVAL Definition must contain a criteria unless the definition is a deprecated definition.</sch:assert>
+                    </sch:rule>
+                </sch:pattern>
+            </xsd:appinfo>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element ref="ds:Signature" minOccurs="0" maxOccurs="1"/>
+            <xsd:element name="metadata" type="oval-def:MetadataType">
+                <!-- xsd:unique name="UniqueAffectedFamily">
+                    <xsd:annotation>
+                        <xsd:documentation>Each affected element must have a unique family attribute value.</xsd:documentation>
+                    </xsd:annotation>
+                    <xsd:selector xpath="oval-def:affected"/>
+                    <xsd:field xpath="@family"/>
+                </xsd:unique -->            
```

Comment:
What's the issue here for why this is commented out?  Unless I'm missing something, it seems no better or worse than the current validation for the old definition type.

Reply:
@johnulmer-oval posted his answer on it back on the issue #181 (he's new to dealing with PR's and Github),  I'll re-post his comments here for easier reading.   Simple answer is that we just need to change them from being unique to elements, as they can no longer be validated as unique.

From @johnulmer-oval :
Have modified oval-definitions-schema.xsd in the '181-update-structure-of-oval-definitions-to-support-encapsulated-defintions...' branch to support encapsulated_definition(s).

Remaining to be addressed.
1 - In the EncapsulatedDefinitionType, the 'xsd:unique name="UniqueAffectedFamily"' is commented out. Need to figure out best way to enforce this requirement.
2 - Review that the modified 'testKeyRef' checks are appropriate.

An example of content employing an encapsulate_definition is attached.

Review and comment is requested.
[simplest_oval6_test.txt](https://github.com/user-attachments/files/18131615/simplest_oval6_test.txt)

As OVAL processing is unchanged from an OVAL 5.11 'definition' to an 'encapsulated_definition', OVAL results are typical.

@[johnulmer-oval](https://github.com/johnulmer-oval)
Member
johnulmer-oval commented [4 hours ago](https://github.com/OVAL-Community/OVAL/issues/181#issuecomment-2549592737)
RE: the 1st 'remaining' issue mentioned in my previous - the 'xsd:unique name="UniqueAffectedFamily"' in the EncapsulatedDefinitionType

When making the EncapsualteDefinitionType, I used the DefinitionType as a template. That brought along the metadata chunk and with it the xsd:unique for the UniqueAffectedFamily. This 'xsd:unique' cannot exist twice with the same element name. So, I'm trying to understand the intended use of the 'family'. After surfing as much documentation as I can find and some content streams, it appears to be mainly a reporting label. If this is the case, I think we can change the xsd:unique to a simple element and the non-uniqueness is eliminated. But, 'would appreciate any input others might have regarding the actual in practice use of the 'family'.

Reply:
For all the content we've seen all the definitions in a given file have the same family.  I can only recall one time seeing a file with both family=unix definitions and some family=windows definitions in the same file, and even in that file each definition only had one family.  It sounds like @johnulmer-oval found a different solution but worst case I don't think it's exceptionally important the schema enforce anything for this attribute.